### ISSUE TITLE
Accordion header usability

### ISSUE THREAD

**Issue by Herst** [twbs/bootstrap#23853]
Apparently since accordion pane headers look like buttons (now more than ever), it has been shown in our user testing that some users are attempting to click the accordion headers first.

Is there any good reason against making the whole header clickable?

Some examples of other web UI frameworks where the whole accordion header is clickable:
* <https://jqueryui.com/accordion/>
* <http://materializecss.com/collapsible.html>
* <https://getuikit.com/docs/accordion>
* <https://material.angular.io/components/expansion/examples>
* <https://www.uplabs.com/posts/material-expansion-panels>
* <https://b-3po.github.io/angular-material-expansion-panel/#/group>
* <https://rijulaggarwal16.github.io/material-expansion-panel/>

**Comment by Herst** [comment]
Examples of other web UI frameworks there the whole accordion headers are clickable:
* <https://jqueryui.com/accordion/>
* <http://materializecss.com/collapsible.html>
* <https://getuikit.com/docs/accordion>

**Comment by Ruffio** [comment]
That would be so nice.
As I get it, then in v. 4. there is no 'accordion' but a 'collapse', but that doesn't have it either: https://getbootstrap.com/docs/4.0/components/collapse/

**Comment by pkozlowski-opensource** [comment]
On the other hand people might want to put other clickable elements in a header (one not causing expand / collapse) - ex. a close button.

Somehow it seems like we should have 2 areas in the accordion / tab header.

**Comment by morrissey-ingenious** [comment]
You can easily do this with the current implementation see below example of a card (looks like a v3 accordion).  You place the data elements on the card-header

```
<div class="card">
    <div class="card-header" id="headingOne" role="tab" data-toggle="collapse" aria-expanded="false" aria-controls="collapseOne" data-target="#collapseOne">
      <h5 class="mb-0">
        <a class="collapsed">
          Collapsible Group Item #1
        </a>
      </h5>
    </div>

    <div id="collapseOne" class="collapse" role="tabpanel" aria-labelledby="headingOne" data-parent="#accordion" style="">
      <div class="card-body">
        Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. 3 wolf moon officia aute, non cupidatat skateboard dolor brunch. Food truck quinoa nesciunt laborum eiusmod. Brunch 3 wolf moon tempor, sunt aliqua put a bird on it squid single-origin coffee nulla assumenda shoreditch et. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident. Ad vegan excepteur butcher vice lomo. Leggings occaecat craft beer farm-to-table, raw denim aesthetic synth nesciunt you probably haven't heard of them accusamus labore sustainable VHS.
      </div>
    </div>
  </div>
```

**Comment by gijsbotje** [comment]
imo it has to be an 'a' element due to accessibility reasons
what you can do is add `display: block;` to the a element and it will stretch accros the header

examples with "icons"
https://codepen.io/gijsbotje/pen/XaoYGY

**Comment by morrissey-ingenious** [comment]
@gijsbotje you can easily add `aria-roll="button"` to my example to resolve any accessibility issues.  Your idea of adding `display:block` is also acceptable that would be accomplished with the class `d-block` in v4.

Either way I think this issue can be closed.

**Comment by Herst** [comment]
> Either way I think this issue can be closed.

How to do it could be mentioned in the docs.

**Comment by XhmikosR** [comment]
PR is welcome.

**Comment by Herst** [comment]
I was about to add a PR but I am still not sure what exactly to mention in the docs. While like @gijsbotje solution with the icons a lot but the code changes needed a are bit too much for the documentation IMHO and I think this would be as a part of bootstrap.

So I guess probably a single line about adding `d-block` to the link will be enough. BTW, this way it's still not like a button as not the whole space is clickable but in this case it could be combined with @morrissey-ingenious's solution by adding the `data-toggle="collapse"` and `href` additionally also to the header `<div>`, what do you think?

> _you can easily add `aria-roll="button"` to my example to resolve any accessibility issues_

I don't agree. One should also think about handicapped users not using screen readers or other people first and foremost using the keyboard and in this case proper links are still better because they can be <kbd>Tab</kbd>ed to.

**Comment by Herst** [comment]
Worth mentioning: Now there is also `.stretched-link` for making links fill their parents.

**Comment by ysds** [comment]
Fixed in #29782
